### ISSUE TITLE
Automatically generated points (points along geometry) cannot be used for TIN Interpolation 

### ISSUE THREAD

**Issue by DanielHorst** [qgis/QGIS#39030]
Hi everyone, 

i am currently working on creating digital elevation models based on planned heights from construction projects - these DEMS are then used for flood modelling in the respective areas. To smooth out the interpolation I use structural lines, break lines (for curbs or walls) as well as a lot of height points which I generate automatically using the function "Geometry along lines".

The initial (auxiliary) lines contain: 
- Elevation at start of the line
- Elevation at end of the line
- Height difference between start and end 
- Length of the line (automatically calculated using §length in field calculator)
- Slope = Calculated from Height difference and length

After creating the running the algorithm to generate the points, the height of each point is then calculated using the starting height, slope and distance to starting point (automatically generated by algorithm): ("Start" - ("Slope" * "distance")

**Now the problem:** 

When I use an interpolation distance (points along geometry) smaller than 60cm the TIN Interpolation, to create the elevation model, does not work for some coordinate systems (e.g. EPSG:2397). For points generated in EPSG:25833 it usually works with an interpolation (points along geometry) distance around 10cm. I attached a screenshot of the python error I frequently get and a sample shape file. 

![Clipboard - 14  September 2020 10_02](https://user-images.githubusercontent.com/71876175/94252569-66fd4000-ff24-11ea-8f91-9fa4c5333867.png)

[Bug report.zip](https://github.com/qgis/QGIS/files/5281960/Bug.report.zip)

Anyone has an idea why this I get these faulty results?

Thanks for your comments and help

**Comment by roya0045** [comment]
So you're saying that only EPSG:2397 causes a problem?

**Comment by DanielHorst** [comment]
No, I have also tried it with different coordinate systems.. not methodologically though. These are just the coordinate systems I am working on for the projects.

**Comment by lbartoletti** [comment]
The algorithm tries to count the number of rows and columns by performing a calculation on the area and size of the pixel.

At line 296, there is a type conversion problem. BTW, the same problem can occurred with the next line.

`self.extent.height() / cell_size` returns (I think) "inf" and round(inf) is not possible (inf is a float, and int cannot have inf value)

So, it seems there is a problem with the extent.

Can you send me the DEM and parameters used? Thanks
